### ISSUE TITLE
SurvSHAP(t) values for cumulative hazard function

### ISSUE THREAD

**Issue by mari-sem** [MI2DataLab/survshap#3]
Dear authors,

firstly, thank you very much for providing the opportunity to explain survival models with SHAP values! 

I would like to calculate SHAP values based on the attribution to the cumulative hazard function. I was thinking that setting predict_cumulative_hazard_function to True for the object SurvivalModelExplainer would mean calculating SHAP for the chf, but no matter how I vary the parameters predict_cumulative_hazard_function and predict_survival_function the resulted SHAP values do not seem to change. 

In the code I saw the opportunity to set the function_type to chf for the PredictSurvSHAP object, but even though SurvivalModelExplainer seems to call PredictSurvSHAP I found no way of setting function_type via SurvivalModelExplainer.

Could you please tell me how can I produce SHAP values for chf? Am I right that current explanations are based on the attribution to survival function?

Kind Regards,
Mariia

**Comment by krzyzinskim** [comment]
Dear Mariia,
thank you for your interest in SurvSHAP(t) method.

You are right, by default, current explanations are based on the attribution to survival function. However, from the technical side, nothing prevents the use of the cumulative hazard function. As you've noticed, this was already available on a local level before (as we focused on local explanations in the article).

Now, I have also added this option on the global level. Just create a ModelSurvSHAP object and set the `function_type` to`"chf"`, i.e., `survshap_global = ModelSurvSHAP(function_type="chf")`.

By default, the model's method called `predict_cumulative_hazard_function` is used in this case, which is compatible with the models implemented in the `scikit-survival` package. I'm going to add support for other methods in the coming days.

Kind Regards,
Mateusz
